edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerConfig.cs

```diff
@@ -15,10 +15,27 @@ public class DockerConfig : IEquatable<DockerConfig>
     {
         // This is not the actual docker image regex, but a less strict version.
         const string ImageRegexPattern = @"^(?<repo>([^/]*/)*)(?<image>[^/:]+)(?<tag>:[^/:]+)?$";
+        // Match for an environment variable in the official POSIX format $ + "letter" + "letter|digit"
+        // Then the port has to be specified with ":"
+        const string ImageUpstreamRegexPattern = @"^\$(?<host>[a-zA-Z_][a-zA-Z0-9_]*?)(?<path>:.*)";
 
         static readonly Regex ImageRegex = new Regex(ImageRegexPattern);
+        static readonly Regex ImageUpstreamRegex = new Regex(ImageUpstreamRegexPattern);
         readonly CreateContainerParameters createOptions;
 
+        internal interface IEnvironmentWrapper
```

Comment:
nit: move to its own file for IEnvrionmentWrapper and EnvironmentWrapper.  Or at least move out from DockerConfig class.